modules/fedimint-lnv2-client/src/send_sm.rs

```diff
@@ -93,11 +93,11 @@ impl State for SendStateMachine {
                 vec![
                     StateTransition::new(
                         Self::gateway_send_payment(
-                            self.common.gateway_api.clone(),
+                            self.common.gateway_api.clone().unwrap(),
```

Comment:
nit: can we use expect here and below?

Reply:
More fundamentally: why is it safe to always unwrap here? Supposedly you want to set the value to `None` in some instances, wouldn't this require special-casing here? Is that change just deferred?